### ISSUE TITLE
Include/Exclude tags from annotations.

### ISSUE THREAD

**Issue by cyriltovena** [grafana/grafana#29536]
**What would you like to be added**:

When doing a Loki query for annotations, it might have too many labels and I may want to select or filter some to show on annotations.

**Why is this needed**:

Loki doesn't support a way to remove labels, not sure it will have that in the future. It seems relatively safe to add.

Example:

![image](https://user-images.githubusercontent.com/1053421/100878482-413a6d00-3478-11eb-96ea-2e5ce56f835d.png)

Here I might want to keep only 2 labels out of those.

**Comment by Elfo404** [comment]
Hi @cyriltovena 
Is there a specific use-case for labels to be removed? 
As far as I understood you only want to hide them from the UI and not have a way to exclude them from the query itself, is this right?

**Comment by cyriltovena** [comment]
Yes that's correct, there's no way to do that with Loki anyway.

I just want to select only what I want to show in the annotation.

**Comment by conorevans** [comment]
@Elfo404 @cyriltovena I could have a look at this. Do you have an idea of how the feature would work? I figure clicking the label to hide/show its value is a good approach? It's consistent with the Graph panel where clicking on a data point can hide/show it.

**Comment by ivanahuckova** [comment]
Hello @conorevans, I personally think it would make sense to implement this in similar way as with Prometheus where we have a inputs where we can specify tags we would like to be included. 
![image](https://user-images.githubusercontent.com/30407135/112458940-9b2fad00-8d5d-11eb-8dfc-95cdf5a8c676.png)

In Loki we take all [labels and push them as tags](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/loki/datasource.ts#L513). @cyriltovena What do you think should be the default behaviour, if no tags would be specified? Show them all or show none?

**Comment by conorevans** [comment]
Ah yes, that seems better @ivanahuckova. I thought the request was from within the panel itself.

Also, as an aside, I noticed that labels with blank values (e.g. `"proxy_protocol_addr": ""`) are still being assigned a tag (see the blank purplish tag third from the top on the far left). I reckon that probably should not be the case. Should I make an issue for that?

**Comment by ivanahuckova** [comment]
> I reckon that probably should not be the case. 

Yeah, I totally agree, empty label values should be definitely omitted. So feel free to either create issue. Or if you are interested, you can create PR as this seems like a single [line fix in here](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/loki/datasource.ts#L514). You just need to filter out empty strings and add a simple test.
`tags.push.apply(tags, [...new Set (Object.values(field.labels).filter((label:string) => label != '' ).map((label:string) => label.trim()))]);`
But creating issue is also awesome!

**Comment by Mario-Hofstaetter** [comment]
@cyriltovena Sorry for being off-topic here, a question regarding your first screenshot:
Should so many labels not be avoided in loki? Or are those parsed with `| json` on the fly?

**Comment by cyriltovena** [comment]
> @cyriltovena Sorry for being off-topic here, a question regarding your first screenshot:
> Should so many labels not be avoided in loki? Or are those parsed with `| json` on the fly?

yes they are on the fly label created by json 😱 don't index those !!!!!!!
